mParticle-Google-Analytics-Firebase/MPKitFirebaseAnalytics.m

```diff
@@ -0,0 +1,340 @@
+    NSMutableDictionary<NSString *, id> *parameters = [[NSMutableDictionary alloc] init];
+    
+    if (product.quantity) {
+        [parameters setObject:product.quantity forKey:kFIRParameterQuantity];
+    }
+    if (product.sku) {
+        [parameters setObject:product.sku forKey:kFIRParameterItemID];
+    }
+    if (product.name) {
+        [parameters setObject:product.name forKey:kFIRParameterItemName];
+    }
+    if (product.category) {
+        [parameters setObject:product.category forKey:kFIRParameterItemCategory];
+    }
+    if (product.price) {
+        [parameters setObject:@(product.price.doubleValue * product.quantity.doubleValue) forKey:kFIRParameterValue];
+        [parameters setObject:product.price forKey:kFIRParameterPrice];
+    }
+    if (commerceEvent.currency) {
+        [parameters setObject:commerceEvent.currency forKey:kFIRParameterCurrency];
+    }
+    if (value) {
+        [parameters setObject:value forKey:kFIRParameterValue];
+    }
+    if (commerceEvent.checkoutStep) {
+        [parameters setObject:@(commerceEvent.checkoutStep) forKey:kFIRParameterCheckoutStep];
+    }
+    if (commerceEvent.checkoutOptions) {
+        [parameters setObject:commerceEvent.checkoutOptions forKey:kFIRParameterCheckoutOption];
+    }
+    if (commerceEvent.transactionAttributes.transactionId) {
+        [parameters setObject:commerceEvent.transactionAttributes.transactionId forKey:kFIRParameterTransactionID];
+    }
+    if (commerceEvent.transactionAttributes.tax) {
+        [parameters setObject:commerceEvent.transactionAttributes.tax forKey:kFIRParameterTax];
+    }
+    if (commerceEvent.transactionAttributes.shipping) {
+        [parameters setObject:commerceEvent.transactionAttributes.shipping forKey:kFIRParameterShipping];
+    }
+    if (commerceEvent.transactionAttributes.couponCode) {
+        [parameters setObject:commerceEvent.transactionAttributes.couponCode forKey:kFIRParameterCoupon];
+    }
+
+    return parameters;
+}
+
+- (NSString *)userIdForFirebase:(FilteredMParticleUser *)currentUser {
+    NSString *userId = currentUser.userId.stringValue;
+    if (currentUser != nil && self.configuration[kMPFIRUserIdFieldKey] != nil) {
+        NSString *key = self.configuration[kMPFIRUserIdFieldKey];
+        if ([key isEqualToString:kMPFIRUserIdValueCustomerID] && currentUser.userIdentities[@(MPUserIdentityCustomerId)] != nil) {
+            userId = currentUser.userIdentities[@(MPUserIdentityCustomerId)];
+        } else if ([key isEqualToString:kMPFIRUserIdValueEmail] && currentUser.userIdentities[@(MPUserIdentityEmail)] != nil) {
+            userId = currentUser.userIdentities[@(MPUserIdentityEmail)];
+        } else if ([key isEqualToString:kMPFIRUserIdValueMPID] && currentUser.userId != nil) {
+            userId = currentUser.userId != 0 ? [currentUser.userId stringValue] : @"0" ;
```

Comment:
Can you return nil here, and then update the consumers of `userIdForFirebase` to handle nil (and just don't set the ID at all)? This will result in a hotkey/bad data downstream.